nixos/modules/services/networking/icecream/daemon.nix

```
@@ -0,0 +1,165 @@
+        '';
+      };
+
+      noRemote = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Prevent jobs from other nodes being scheduled on this daemon.
+        '';
+      };
+
+      schedulerHost = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = ''
+          Explicit scheduler hostname, useful in firewalled environments.
+        '';
+      };
+
+      maxProcesses = mkOption {
+        type = types.nullOr types.ints.u16;
+        default = null;
+        description = ''
+          Maximum number of compile jobs started in parallel for this daemon.
+        '';
+      };
+
+      nice = mkOption {
+        type = types.nullOr types.int;
+        default = null;
+        description = ''
+          The level of niceness to use. Defaults to 5 when null.
+        '';
+      };
+
+      hostname = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = ''
+          Hostname of the daemon in the icecream infrastructure.
+        '';
+      };
+
+      package = mkOption {
+        default = pkgs.icecream;
+        defaultText = "pkgs.icecream";
+        type = types.package;
+        description = "Icecream package to use.";
+      };
+
+      extraConfig = mkOption {
+        type = types.str;
+        default = "";
+        description = "Additional command line parameters";
+        example = "-v";
+      };
+    };
+  };
+
+  ###### implementation
+
+  config = mkIf cfg.enable {
+    systemd.tmpfiles.rules = [
+      "d '${cfg.baseDir}' - ${user} ${group} - -"
+    ];
+
+    users.users.icecc = {
+      description = "Icecc daemon user";
+      group = group;
+      uid = config.ids.uids.icecc;
+      home = cfg.baseDir;
+      createHome = true;
+    };
+
+    users.groups.icecc = {
+      gid = config.ids.gids.icecc;
+    };
+
+    networking.firewall.allowedTCPPorts = mkIf cfg.openFirewall [ 10245 ];
+    networking.firewall.allowedUDPPorts = mkIf cfg.openBroadcast [ 8765 ];
+
+    systemd.services.icecc-daemon = {
+      description = "Icecream compile daemon";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+
+      serviceConfig = {
+        ExecStart = concatStringsSep " " [
+          "${getBin cfg.package}/bin/iceccd"
+          "-b ${cfg.baseDir}"
+          "${optionalString (cfg.netName != null) "-n ${cfg.netName}"}"
+          "${optionalString (cfg.cacheLimit != null) "--cache-limit ${toString cfg.cacheLimit}"}"
+          "${optionalString (cfg.maxProcesses != null) "-m ${toString cfg.maxProcesses}"}"
+          "${optionalString (cfg.hostname != null) "-N ${cfg.hostname}"}"
+          "${optionalString cfg.noRemote "--no-remote"}"
+          "${optionalString (cfg.nice != null) "--nice ${toString cfg.nice}"}"
+          "${optionalString (cfg.schedulerHost != null) "-s ${cfg.schedulerHost}"}"
+          "${cfg.extraConfig}"
```

```suggestion
          cfg.extraConfig
```
The same type of simplification applies above.